packages/roosterjs-content-model/lib/modelApi/selection/collapseTableSelection.ts

```diff
@@ -0,0 +1,23 @@
+import { addSegment } from '../common/addSegment';
+import { ContentModelTableCell } from '../../publicTypes/block/group/ContentModelTableCell';
+import { createSelectionMarker } from '../creators/createSelectionMarker';
+import { TableSelectionCoordinates } from './setSelectionToTable';
+
+/**
+ * @internal
+ */
+export function collapseTableSelection(
+    cells: ContentModelTableCell[][],
+    selection: TableSelectionCoordinates
+) {
+    const { firstCol, firstRow, lastCol, lastRow } = selection;
+
+    for (let row = 0; row < cells.length; row++) {
```

Comment:
You just need to add a selection marker to the first selected cell if exist:

const cell =cells[firstRow]?.[firstCol];

if (cell) { addSegment(...}

Reply:
Ah yes, good point. 😅